src/backend/base/langflow/components/astra_assistants/astra_assistant_manager.py

```diff
@@ -0,0 +1,134 @@
+import asyncio
+
+from astra_assistants.astra_assistants_manager import AssistantManager
+
+from langflow.components.astra_assistants.util import (
+    get_patched_openai_client,
+    litellm_model_names,
+    tool_names,
+    tools_and_names,
+)
+from langflow.custom.custom_component.component_with_cache import ComponentWithCache
+from langflow.inputs import DropdownInput, MultilineInput, StrInput
+from langflow.schema.message import Message
+from langflow.template import Output
+
+
+class AstraAssistantManager(ComponentWithCache):
+    display_name = "Astra Assistant Manager"
+    description = "Manages Assistant Interactions"
+    icon = "bot"
+
+    inputs = [
+        StrInput(
+            name="instructions",
+            display_name="Instructions",
+            info="Instructions for the assistant, think of these as the system prompt.",
+        ),
+        DropdownInput(
+            name="model_name",
+            display_name="Model Name",
+            advanced=False,
+            options=litellm_model_names,
+            value="gpt-4o-mini",
+        ),
+        DropdownInput(
+            display_name="Tool",
+            name="tool",
+            options=tool_names,
+        ),
```

Comment:
Ideally, the tools would be Components. 

What if these tools work as default ones?

Reply:
Maybe there's a way to do both, pick from the dropdown if the tools are simple and don't have any configuration or add an edge from a component for more complex tools. We would probably also want a way to accept langchain basetool tools for assistants dynamically.

Reply:
You should also be able to pass more than one tool and let it decide. Is there a multi-select input in langflow?